recipes/openai_textcat_multilabel.py

```diff
@@ -0,0 +1,527 @@
+                headers=headers,
+                json={
+                    "model": self.model,
+                    "prompt": prompts,
+                    "temperature": self.openai_temperature,
+                    "max_tokens": self.openai_max_tokens,
+                },
+            ),
+            n=self.openai_n,
+            timeout_s=self.openai_timeout_s,
+        )
+        r.raise_for_status()
+        responses = r.json()
+        return [responses["choices"][i]["text"] for i in range(len(prompts))]
+
+    def _parse_response(self, text: str) -> Dict[str, str]:
+        """Interpret OpenAI's TextCat response. It's supposed to be
+        a list of lines, with each line having the form:
+
+        answer: <accept/reject for binary or the label for multilabel>
+        reason: <string>
+
+        However, there's no guarantee that the model will give
+        us well-formed output. It could say anything, it's an LM.
+        So we need to be robust.
+        """
+        output = {}
+        for line in text.strip().split("\n"):
+            if line and ":" in line:
+                key, value = line.split(":", 1)
+                output[key.strip().lower()] = value.strip()
+        return output
+
+
+@prodigy.recipe(
+    # fmt: off
+    "textcat_multilabel.openai.correct",
+    input_path=("Path to jsonl data to annotate", "positional", None, Path),
+    labels=("Labels (comma delimited)", "positional", None, lambda s: s.split(",")),
+    lang=("Language to initialize spaCy model", "option", "l", str),
+    model=("GPT-3 model to use for completion", "option", "m", str),
+    batch_size=("Batch size to send to OpenAI API", "option", "b", int),
+    segment=("Split sentences", "flag", "S", bool),
+    examples_path=("Examples file to help define the task", "option", "e", Path),
+    prompt_path=("Path to jinja2 prompt template", "option", "p", Path),
+    max_examples=("Max examples to include in prompt", "option", "n", int),
+    exclusive_classes=("Make the classification task exclusive", "flag", "E", bool),
+    verbose=("Print extra information to terminal", "flag", "v", bool),
+    # fmt: on
+)
+def textcat_multilabel_openai_correct(
+    dataset: str,
+    input_path: Path,
+    labels: List[str],
+    lang: str = "en",
+    model: str = "text-davinci-003",
+    batch_size: int = 10,
+    segment: bool = False,
+    examples_path: Optional[Path] = None,
+    prompt_path: Path = DEFAULT_PROMPT_PATH,
+    max_examples: int = 2,
+    exclusive_classes: bool = False,
+    verbose: bool = False,
+):
+    """Get bulk TextCat suggestions from an OpenAI API, using zero-shot or
+    few-shot learning.  The results can then be corrected using the
+    `textcat.manual` recipe.
+
+    Here, we use ChatGPT to suggest if a particular text talks about a recipe or
+    not. You can use the parameter `--chatgpt-bias` (float, 0 to 1 inclusive) to
+    set how much we prefer seeing examples that ChatGPT thinks as recipes.
```

Comment:
I think this is partially the text from the fetch recipe and it also still references `recipes`.